### PR TITLE
Subscription storage async

### PR DESCRIPTION
**Pull request by andreasohlund**

### CODE REVIEW

**Comment by andreasohlund** [comment]
@Particular/tf-asyncawait please review

File: src/NServiceBus.RavenDB.Tests/SubscriptionStorage/When_receiving_a_subscription_message.cs [hunk 1]
```diff
@@ -22,7 +23,7 @@ public void A_subscription_entry_should_be_added_to_the_database()
 
         var storage = new SubscriptionPersister(store);
 
-        storage.Subscribe(clientEndpoint, messageTypes, new ContextBag());
+        await storage.Subscribe(clientEndpoint, messageTypes, new ContextBag()).ConfigureAwait(false);
```
Comment (timbussmann) [inline src/NServiceBus.RavenDB.Tests/SubscriptionStorage/When_receiving_a_subscription_message.cs]: I thought we don't use `ConfigureAwait` in the acceptance tests?

Comment (andreasohlund) [inline src/NServiceBus.RavenDB.Tests/SubscriptionStorage/When_receiving_a_subscription_message.cs]: Missed that, will fix

Comment (andreasohlund) [inline src/NServiceBus.RavenDB.Tests/SubscriptionStorage/When_receiving_a_subscription_message.cs]: fixed

**Pull request merged**